app/services/sort_stories.rb

```diff
@@ -0,0 +1,25 @@
+class SortStories
+  attr_reader :stories
+
+  def initialize(ordered_ids)
+    @ordered_ids = ordered_ids
+    @stories = Story.find(@ordered_ids)
+  end
+
+  def call
+    @stories.each do |story|
+      yield(story) if block_given?
+      update_position(story)
+    end.sort_by!(&:position)
```

Comment:
You don't need this bang here since you're returning a new sorted list when you call the `.call` method.
If you want to keep it, for cases when you do something like:
```ruby
@sorted_list = SortStories.new(args).call()
@sorted_list.stories # => returns the sorted list
```

you should rename the @stories name to something more meaningful(referencing the sorting)

But I suggest you to just remove the bang.

Reply:
Also, why didn't you use a `map` instead of the `each`?